src/System.Resources.Extensions/src/System/Resources/Extensions/DeserializingResourceReader.cs

```diff
@@ -0,0 +1,204 @@
+                        Stream stream;
+
+                        if (_store.BaseStream is UnmanagedMemoryStream ums)
+                        {
+                            // For the case that we've memory mapped in the .resources
+                            // file, just return a Stream pointing to that block of memory.
+                            unsafe
+                            {
+                                stream = new UnmanagedMemoryStream(ums.PositionPointer, length, length, FileAccess.Read);
+                            }
+                        }
+                        else
+                        {
+
+                            byte[] bytes = _store.ReadBytes(length);
+                            // Lifetime of memory == lifetime of this stream.
+                            stream = new MemoryStream(bytes, false);
+                        }
+
+                        value = Activator.CreateInstance(type, new object[] { stream });
+                        break;
+                    }
+                default:
+                    throw new BadImageFormatException(SR.BadImageFormat_TypeMismatch);
+            }
+
+            if (value.GetType() != type)
+                throw new BadImageFormatException(SR.Format(SR.BadImageFormat_ResType_SerBlobMismatch, type.FullName, value.GetType().FullName));
+
+            return value;
+        }
+
+        private static bool CompareNames(string typeName1, string typeName2)
+        {
+            // First, compare type names
+            int comma1 = typeName1.IndexOf(',');
+            int comma2 = typeName2.IndexOf(',');
+            if (comma1 != comma2)
+                return false;
+
+            // both are missing assembly name, compare entire string as type name
+            if (comma1 == -1)
+                return string.Equals(typeName1, typeName2, StringComparison.Ordinal);
+
+            // compare the type name portion
+            ReadOnlySpan<char> type1 = typeName1.AsSpan(0, comma1);
+            ReadOnlySpan<char> type2 = typeName2.AsSpan(0, comma2);
+            if (!type1.Equals(type2, StringComparison.Ordinal))
+                return false;
+
+            // Now, compare assembly display names (IGNORES VERSION AND PROCESSORARCHITECTURE)
+            // also, for  mscorlib ignores everything, since that's what the binder is going to do
+            while (Char.IsWhiteSpace(typeName1[++comma1]))
+                ;
+            while (Char.IsWhiteSpace(typeName2[++comma2]))
+                ;
+
+            // case insensitive
+            AssemblyName an1 = new AssemblyName(typeName1.Substring(comma1));
+            AssemblyName an2 = new AssemblyName(typeName1.Substring(comma2));
+            if (!string.Equals(an1.Name, an2.Name, StringComparison.OrdinalIgnoreCase))
+                return false;
+
+            // to match IsMscorlib() in VM
+            if (string.Equals(an1.Name, "mscorlib", StringComparison.OrdinalIgnoreCase))
```

Comment:
>mscorlib [](start = 41, length = 8)

Is this valid check anymore?

Reply:
It comes from here: https://github.com/dotnet/coreclr/blob/e6034d903f2608445a3f66e3694f461fad7b8b88/src/System.Private.CoreLib/shared/System/Resources/ResourceManager.cs#L591-L592

And originally from desktop code here: https://referencesource.microsoft.com/#mscorlib/system/resources/resourcemanager.cs,948

I believe the reason for this is:
> for  mscorlib ignores everything, since that's what the binder is going to do

So I believe they are saying that the binder ignores culture and key-token when loading a type that claims it is in MSCorlib.  This is still relevant as this code will run on both .NETCore and desktop.  Even on .NETCore we're still doing this check so I'm reluctant to remove it.  Given these strings come from user assemblies and we're trying to be compatible with what we previously allowed I don't feel comfortable removing/changing this without a lot of data saying its safe to do so.

Reply:
do we need to check for System.Private.CoreLib too then?

Reply:
`System.Private.CoreLib` is not supposed to show up in any wire-serialized formats.

Reply:
ok, thanks. it is nice to know that.